src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -151,6 +154,75 @@
     </ItemGroup>
   </Target>
 
+  <!-- #################################### -->
+  <!-- Generate BuildVersion.props -->
+  <!-- #################################### -->
+  <Target Name="CreateOrUpdateCurrentVersionFile" Condition="'$(SkipVersionGeneration)'!='true'" DependsOnTargets="GenerateCurrentVersion">
+    <Error Condition="'$(OverrideBuildNumberMajor)'==''" Text="Error when generating current version." />
+
+    <ItemGroup>
+      <CurrentVersionLines Include="%3C%3Fxml version=%221.0%22 encoding=%22utf-8%22%3F%3E" />
+      <CurrentVersionLines Include="%3C!-- This is a generated file. To re-generate it, run build.cmd --%3E" />
```

Comment:
I think I would exclude the "regenerate it" step because it might be different across repos.